### ISSUE TITLE
After following Usage in Readme.md: SCALE_FUNCTIONS[type] is not a function

### ISSUE THREAD

**Issue by mreishus** [uber/react-vis#308]
After following the usage directions in Readme.md (essentially `yarn add react-vis`, `yarn add react-addons-shallow-compare`, setting up a few imports and adding the following to a component: )

```
<XYPlot
  width={300}
  height={300}>
  <HorizontalGridLines />
  <LineSeries
    data={[
      {x: 1, y: 10},
      {x: 2, y: 5},
      {x: 3, y: 15}
    ]}/>
  <XAxis />
  <YAxis />
</XYPlot>
```

The graph fails to render and I get the console error message `Uncaught TypeError: SCALE_FUNCTIONS[type] is not a function`.  I'm using webpack2, presets for react and es2015.. what I thought was a standard setup.

The line that is failing is here: https://github.com/uber/react-vis/blob/d96abc8841c2a6312aa/src/utils/scales-utils.js#L146

If I go into my node_modules directory and add a `debugger` statement just before it, I can see that my `SCALE_FUNCTIONS` only has `literal` defined, and that `type` is `"ordinal"`.

![Debugger](http://i.imgur.com/dvQyqe8.png)

Has anyone seen this before or have any ideas what this could be?  Unfortunately my workplace blocks gist and other services so I can't easily drop my yarn lock file.

Here's the stacktrace:
```
Uncaught TypeError: SCALE_FUNCTIONS[type] is not a function
    at _getScaleFnFromScaleObject (eval at <anonymous> (app.js:1598), <anonymous>:187:36)
    at _adjustCategoricalScale (eval at <anonymous> (app.js:1598), <anonymous>:493:17)
    at getScaleObjectFromProps (eval at <anonymous> (app.js:1598), <anonymous>:531:12)
    at getAttributeScale (eval at <anonymous> (app.js:1598), <anonymous>:543:21)
    at GridLines.render (eval at <anonymous> (app.js:1945), <anonymous>:146:54)
    at eval (eval at <anonymous> (vendor.js:6114), <anonymous>:796:21)
    at measureLifeCyclePerf (eval at <anonymous> (vendor.js:6114), <anonymous>:75:12)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (eval at <anonymous> (vendor.js:6114), <anonymous>:795:25)
    at ReactCompositeComponentWrapper._renderValidatedComponent (eval at <anonymous> (vendor.js:6114), <anonymous>:822:32)
    at ReactCompositeComponentWrapper.performInitialMount (eval at <anonymous> (vendor.js:6114), <anonymous>:362:30)
```

react-vis is version 1.0.1, my react version is 15.4.2, webpack is 2.3.1, babel-core is 6.24.  Babel presets are react, es2015 with these plugins:
```
        ["syntax-dynamic-import"],
        ["transform-decorators-legacy"],
        ["transform-object-rest-spread"],
        ["transform-react-display-name"],
        ["transform-object-assign"],
        ["react-transform", {
          "transforms": [{
            "transform": "react-transform-hmr",
            "imports": ["react"],
            "locals": ["module"]
          }, {
            "transform": "react-transform-catch-errors",
            "imports": ["react", "redbox-react"]
          }]
        }]
```

**Comment by mreishus** [comment]
Some things I have tried temporarily removing from my config:

- Hot module reloading
- Vendor splitting via common chunk plugin
- babel-polyfill

None of these worked.  Could I just be doing something really stupid like not loading D3?

**Comment by mreishus** [comment]
I found what was causing the issue.  It's this line in my `module.exports` in my webpack config:

```
    devtool:  process.env.NODE_ENV == "production" ? "source-map" : "cheap-module-eval-source-map",
```

Setting the devtool to `cheap-module-eval-source-map` made the library not work.  I tested them all since I was interested:

works
----
eval
eval-source-map
source-map
nosources-source-map
inline-source-map


does not work
--------------
cheap-eval-source-map
cheap-source-map
cheap-module-eval-source-map
cheap-module-source-map

I also found that if my devtool was set to one of the ones that does not work, if I added `console.log('hi')` at the top of my `node_modules/d3-scale/index.js` file, it would start working...

Still not sure what is going on, but the library works for me now.  I'll keep this open just in case someone who works on the library would find this useful.

**Comment by mcnuttandrew** [comment]
@apercu do you have any thoughts on this?

**Comment by Apercu** [comment]
Hum yes, quite strange.

From what @mreishus is describing, it looks like d3-scale is loaded after react-vis, which should not be possible. Might also be due to the `import * as` and the babel presets he has (no stage-0).

**Comment by vjpr** [comment]
For me only `SCALE_FUNCTIONS.literal` is available...

![screen shot 2017-08-21 at 5 23 05 pm](https://user-images.githubusercontent.com/281413/29526254-74971a98-8695-11e7-90de-b7d4d2cbbf2f.png)

**Comment by vjpr** [comment]
If we look at the scope at that breakpoint: 

![screen shot 2017-08-21 at 5 40 59 pm](https://user-images.githubusercontent.com/281413/29527082-f3d85afe-8697-11e7-923a-e0254e3f4a8b.png) 

there are a bunch of `__WEBPACK_IMPORTED_MODULE_1_d3_array__` for each import, but nothing for `d3-scale`.

`d3-scale/index.js` looks like this:

```
export {default as category10} from "./src/category10";
export {default as category20b} from "./src/category20b";
export {default as category20c} from "./src/category20c";
export {default as category20} from "./src/category20";
export {default as cubehelix} from "./src/cubehelix";
export {default as identity} from "./src/identity";
export {default as linear} from "./src/linear";
export {default as log} from "./src/log";
export {default as ordinal} from "./src/ordinal";
export {default as pow, sqrt} from "./src/pow";
export {default as quantile} from "./src/quantile";
export {default as quantize} from "./src/quantize";
export {default as rainbow} from "./src/rainbow";
export {default as threshold} from "./src/threshold";
export {default as time} from "./src/time";
export {default as utcTime} from "./src/utcTime";
```

Is this even valid?

**Comment by vjpr** [comment]
Okay found the issue. There is a mismatch between the `d3-scale` exports and the ones used in `scales-util`.

```
export {default as category10} from "./src/category10";
export {default as category20b} from "./src/category20b";
export {default as category20c} from "./src/category20c";
export {default as category20} from "./src/category20";
export {default as cubehelix} from "./src/cubehelix";
export {default as identity} from "./src/identity";
export {default as linear} from "./src/linear";
export {default as log} from "./src/log";
export {default as ordinal} from "./src/ordinal";
export {default as pow, sqrt} from "./src/pow";
export {default as quantile} from "./src/quantile";
export {default as quantize} from "./src/quantize";
export {default as rainbow} from "./src/rainbow";
export {default as threshold} from "./src/threshold";
export {default as time} from "./src/time";
export {default as utcTime} from "./src/utcTime";
```

```
var SCALE_FUNCTIONS = (_SCALE_FUNCTIONS = {}, _defineProperty(_SCALE_FUNCTIONS, LINEAR_SCALE_TYPE, _d3Scale.scaleLinear), _defineProperty(_SCALE_FUNCTIONS, ORDINAL_SCALE_TYPE, _d3Scale.scalePoint), _defineProperty(_SCALE_FUNCTIONS, CATEGORY_SCALE_TYPE, _d3Scale.scaleOrdinal), _defineProperty(_SCALE_FUNCTIONS, LITERAL_SCALE_TYPE, literalScale), _defineProperty(_SCALE_FUNCTIONS, LOG_SCALE_TYPE, _d3Scale.scaleLog), _defineProperty(_SCALE_FUNCTIONS, TIME_SCALE_TYPE, _d3Scale.scaleTime), _defineProperty(_SCALE_FUNCTIONS, TIME_UTC_SCALE_TYPE, _d3Scale.scaleUtc), _SCALE_FUNCTIONS);
```

**Comment by mcnuttandrew** [comment]
Hey @vjpr,

Can you say more about the mismatch you found? I'm not sure I get it? Also, this is a long shot (🤞)  but this issue might have been resolved by #567
